### PR TITLE
qt 6.5.2

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```
Undefined symbols for architecture x86_64:
  "_png_create_info_struct", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_create_read_struct", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_destroy_read_struct", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_error", referenced from:
      _read_data_from_FT_Stream in sfnt.o
  "_png_get_IHDR", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_get_error_ptr", referenced from:
      _error_callback in sfnt.o
      _read_data_from_FT_Stream in sfnt.o
  "_png_get_io_ptr", referenced from:
      _read_data_from_FT_Stream in sfnt.o
  "_png_get_valid", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_read_end", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_read_image", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_read_info", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_read_update_info", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_expand_gray_1_2_4_to_8", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_filler", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_gray_to_rgb", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_interlace_handling", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_longjmp_fn", referenced from:
      _Load_SBit_Png in sfnt.o
      _error_callback in sfnt.o
  "_png_set_packing", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_palette_to_rgb", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_read_fn", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_read_user_transform_fn", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_strip_16", referenced from:
      _Load_SBit_Png in sfnt.o
  "_png_set_tRNS_to_alpha", referenced from:
      _Load_SBit_Png in sfnt.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

**Comment by fxcoudert** [comment]
Filed a bug upstream for the PNG build failure: https://bugreports.qt.io/browse/QTBUG-115357

File: Formula/qt.rb
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/qt.rb:1]: my bad, I mis-read that ticket.

**Comment by p-linnane** [comment]
Rebased with all the work @iMichka has done.

**Comment by iMichka** [comment]
Canceling as there are already too many failures on macOS. Let's free up the build machines to build something else while we investigate here.

**Comment by p-linnane** [comment]
Looks like patches need to either be removed or rebased.

**Comment by fxcoudert** [comment]
Trying to see if we can temporarily work around the bug by using `-DQT_FEATURE_webengine_system_libpng=OFF`

**Comment by fxcoudert** [comment]
```
  qt-mysql
    * 'revision 1' should be removed
```

Duh. But otherwise seems to build fine. But I want to try https://github.com/Homebrew/homebrew-core/pull/149708 next, to help upstream find a good fix (instead of using a bundled libpng, which is a hack)

**Comment by fxcoudert** [comment]
Unless something unexpected happens, this should be ready to merge. (The alternative approach is not working yet.)

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**